src/auto-evo/AutoEvoExploringTool.cs

```diff
@@ -418,6 +422,29 @@ public void AskExit()
         exitConfirmationDialog.PopupCenteredShrink();
     }
 
+    public Species? GetActiveSpeciesData(uint speciesId)
+    {
+        var gameworld = world.GameProperties.GameWorld;
```

Comment:
I was originally going to comment that this should always look from the latest generation and in all worlds, but now that I've thought about this a bit I think your solution might be more correct.

So just to confirm this doesn't look starting from the latest generation to make sure this doesn't show too new images of species if going back in time, right?

If that is the case then I think starting from the displayed generation is the correct approach, and then there isn't a need to do anything else to this method except fixing "gameworld" to use camelCase.


For reference I was thinking the code could be something like:

```c#
private Species? LookForSpeciesInWorld(GameWorld specificWorld, uint species)
{
    for (int i = max generation from somewhere; i >= 0; --i)
    {
        return species if found
    }
     
      return null;
}

public Species? GetActiveSpeciesData(uint speciesId)
{
     var species = LookForSpeciesInWorld(world, speciesId);
     
     if(species)
           return species;
     
     foreach (var currentWorld in worldList)
     {
         return species if found
     }
     return null;
}
```

Reply:
Yes, it starts looking from the displayed generation in the current world to display the right 'version' of the species.